src/main/java/jcifs/smb/DirFileEntryEnumIterator2.java

```diff
@@ -106,6 +116,30 @@ protected FileEntry open () throws CIFSException {
                 }
             }
 
+            // We hit a symbolic link, parse the error data and resend for the 'real' directory or file path
+            if (cr != null && cr.isReceived() && cr.getStatus() == NtStatus.NT_STATUS_STOPPED_ON_SYMLINK) {
+                try {
+                    byte[] errorData = cr.getErrorData();
+                    Smb2ErrorContextResponse ecr = new Smb2ErrorContextResponse();
+                    int symLinkLength = ecr.readErrorContextResponse(errorData);
+                    log.debug("symLinkLength -> {}", symLinkLength);
+
+                    String substituteName = ecr.getSubstituteName();
+                    log.debug("substituteName -> {}", substituteName);
+
+                    int i = substituteName.indexOf(getParent().getLocator().getShare());
+                    String realPath = substituteName.substring(i + getParent().getLocator().getShare().length());
+                    log.debug("symLinkPath -> {}", uncPath);
+                    log.debug("realPath -> {}", realPath);
+
+                    return open(realPath);
```

Comment:
Normally I don't put a static counter in a method to to control the recursion depth, as it can be problematic, however I found this on SO that appears to be thread safe, not sure if you agree with this though... the one marked as best answer appears to satisfy the requirement. What is your opinion?

https://stackoverflow.com/questions/53729546/set-maximum-recursion-depth-in-java

Reply:
I wasn't thinking about a static variable, this will definitly cause concurrency issues, but more about another parameter for the open method that tracks the current depth and a limit based on that.

Reply:
What has been implemented based on the SO conversation is more robust than having to track a counter whether it be in a parameter to the method or a static counter. It's thread safe and works effectively, you can test it by changing the `maxLevel` to a lower number.